### PR TITLE
Fix: entity localization parent fallback

### PR DESCRIPTION
**Pull request by codebykoi**
When I working with a locale files for my fork, I found it troublesome when working with many different .ftl files, and my builds always were giving me errors about missing .desc and etc, not allowing me to run properly.

this PR its basically a QoL fixes when working with locales.

Keep in mind that its my first ever PR to RobustToolbox, so I may dont know some specific details about the codebase, but I hope this one works for everyone.

### CODE REVIEW

**Comment by codebykoi** [comment]
found another bug, fixing it rn

**Comment by codebykoi** [comment]
alright, added small tests, and added NeverPushInheritance attribute to the fields of EntityPrototype.
Funny enough, the comments were saying they are not supposed to be inherited way before I did the fix

**Comment by codebykoi** [comment]
Test was failing because of the RCD, the recent commit should fixed the edge case when entity had no name at all, it just fallbacks to protoId instead

**Comment by codebykoi** [comment]
I have no idea why test failed on RandomSpecies test. I literalyl have nothing to do with it.

File: Robust.Shared/Prototypes/EntityPrototype.cs [hunk 1]
```diff
@@ -51,12 +52,15 @@ public sealed partial class EntityPrototype : IPrototype, IInheritingPrototype,
         /// </summary>
         /// <seealso cref="Name"/>
         [DataField("name")]
+        [NeverPushInheritance]
```
Comment (slarticodefast) [inline Robust.Shared/Prototypes/EntityPrototype.cs:55]: won't this completely break almost all inherited entity prototype names, descriptions and suffixes in the entire content repo?

Comment (codebykoi) [reply Robust.Shared/Prototypes/EntityPrototype.cs:55]: uh, doesnt break for me. At least on my locale, I can run it on en-US again. one second

Comment (codebykoi) [reply Robust.Shared/Prototypes/EntityPrototype.cs:55]: > won't this completely break almost all inherited entity prototype names, descroptions and suffixes in the entire content repo?

Just checked in local build, the en-US entity names act the same as before

**Comment by codebykoi** [comment]
and here we go, all tests passes

File: Robust.Shared/Prototypes/EntityPrototype.cs
Comment (codebykoi) [inline Robust.Shared/Prototypes/EntityPrototype.cs:34]: its literally says here that fields should not be inherited

File: Robust.Shared/Prototypes/EntityPrototype.cs (hunk 1 above)
Comment (slarticodefast) [reply Robust.Shared/Prototypes/EntityPrototype.cs:55]: I mean for entities with inherited names etc, because this would stop them from being inherited

Comment (codebykoi) [reply Robust.Shared/Prototypes/EntityPrototype.cs:55]: > I mean for entities with inherited names etc, because this would stop them from being inherited

No it doesnt

Content Client_zPAXNmpvCA
Cursor_kw65datjL1

Comment (slarticodefast) [reply Robust.Shared/Prototypes/EntityPrototype.cs:55]: huh, nevermind then
